src/tls/incoming.rs

```diff
@@ -158,6 +160,26 @@ impl MaybeTlsIncomingStream<TcpStream> {
         Ok(())
     }
 
+    // Required for `make check-component-features`
+    #[allow(dead_code)]
+    pub(crate) fn set_keepalive(
```

Comment:
Do you know which features require this function? We would prefer to feature gate it (ie `#[cfg(all(feature = "X", feature = "Y"))]` rather than use this dead code allowance.

Reply:
Agree, I don't like this part here, at all.

My first approach was introducing a `tcp` flag which `tls` depends on, and put the `set_keepalive` behind the `tcp` flag. Unfortunately, that didn't solve the problem, as it still showed as unused code. I feel making this work as intended would require to refactor larger parts of the `tcp` util. Maybe we make a separate ticket for that?

Reply:
@bruceg Refactor or ticket? I lean towards ticket.

Reply:
Looking into that now.

Reply:
Done in 8e21651. Was less invasive than I anticipated, now after the flags have been cleaned up on master.